### PR TITLE
USJ tests and fixes

### PR DESCRIPTION
**Pull request by kavitharaju**
This PR includes:
- Pulls recent testsuite updates from usfm-bible/tcdocs
- Replaces the USJ samples we had generated here using our grammar, with the usx2usj converted ones. This is copied from the [PR to the tcdocs repo](https://github.com/usfm-bible/tcdocs/pull/45).
- Adds a new set of tests to compare the USJs generated using grammar and those in testsuite
- Fixes based on the above tests
   - Grammar changes:
      - Keep \b at paragraph level, instead of as paragraph content
      - Allow footnotes and cross-refs to come within character markers
   - Fixes in USJ generation code:
      - Handling of \b
      - Avoiding empty strings in contents
      - Wrong order of some elements

### CODE REVIEW

File: py-usfm-parser/tests/test_json_conversion.py [hunk 1]
```diff
@@ -116,3 +116,54 @@ def test_output_is_valid_usj(file_path):
     assert not test_parser.errors, test_parser.errors
     usj_dict = test_parser.to_usj()
     validate(instance=usj_dict, schema=USJ_SCHEMA)
+
+def remove_newlines_in_text(usj_dict):
+    '''The test samples in testsuite do not preserve new lines in. But we do in usfm-grammar.
```
Comment (joelthe1) [inline py-usfm-parser/tests/test_json_conversion.py:121]: We should talk about this sometime. Okay for now.

File: py-usfm-parser/tests/test_json_conversion.py [hunk 2]
```diff
@@ -116,3 +116,54 @@ def test_output_is_valid_usj(file_path):
...
+    So removing them just for comparison'''
+    if "content" in usj_dict:
+        for i,item in enumerate(usj_dict["content"]):
+            if isinstance(item, str):
+                usj_dict['content'][i] = item.replace("\n", " ")
+                usj_dict['content'][i] = usj_dict['content'][i].replace("  ", " ")
+                usj_dict['content'][i] = usj_dict['content'][i].replace("     ", " ")
+                continue
+            if "sid" in item and "PSA 09" in item['sid']: # for /usfmjsTests/tstudio/origin.usfm
+                item['sid'] = item['sid'].replace("PSA 091:01", "PSA 91:1")
```
Comment (joelthe1) [inline py-usfm-parser/tests/test_json_conversion.py:131]: Are we stripping away the `0` in the front?

File: py-usfm-parser/tests/test_json_conversion.py (hunk 1 above)
Comment (kavitharaju) [reply py-usfm-parser/tests/test_json_conversion.py:121]: In our usfm-grammar way, we preserve the line breaks and multiple space as it is in input USFM. But the USXs in testsuite do not do that. Hence the USJs generated from those USXs don't have those spaces. I didn't want to remove spaces when we generate USJs. Also thought not to raise this as a problem with testsuite. Hence I handled them like this on the tests to get them pass. That s what's happening here. But we can talk about it.

File: py-usfm-parser/tests/test_json_conversion.py (hunk 2 above)
Comment (kavitharaju) [reply py-usfm-parser/tests/test_json_conversion.py:131]: We are not, in usfm-grammar. But the USXs in testsuite have them stripped in some places, but not all. I am just changing them in test, to get the comparison to pass, as this is known discrepancy. Same thing as with the space handling. That's what is going on here

**Pull request merged**